src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -0,0 +1,431 @@
+            if (name == "GMT" || name == "UTC")
+            {
+                return new TimeZoneInfo(id, TimeSpan.FromSeconds(0), id, name, name, null, disableDaylightSavingTime:true);
+            }
+            if (name.StartsWith("GMT", StringComparison.Ordinal))
+            {
+                return new TimeZoneInfo(id, TimeSpan.FromSeconds(ParseGMTNumericZone(name)), id, name, name, null, disableDaylightSavingTime:true);
+            }
+
+            try
+            {
+                byte[] buffer = AndroidTzDataInstance.GetTimeZoneData(name);
+                return GetTimeZoneFromTzData(buffer, id);
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        // Core logic to retrieve the local system time zone.
+        // Obtains Android's system local time zone id to get the corresponding time zone
+        // Defaults to Utc if local time zone cannot be found
+        private static TimeZoneInfo GetLocalTimeZoneCore()
+        {
+            string? id = Interop.Sys.GetDefaultTimeZone();
+            if (!string.IsNullOrEmpty(id))
+            {
+                TimeZoneInfo? defaultTimeZone = GetTimeZone(id, id);
+
+                if (defaultTimeZone != null)
+                {
+                    return defaultTimeZone;
+                }
+            }
+
+            return Utc;
+        }
+
+        private static TimeZoneInfoResult TryGetTimeZoneFromLocalMachineCore(string id, out TimeZoneInfo? value, out Exception? e)
+        {
+
+            value = id == LocalId ? GetLocalTimeZoneCore() : GetTimeZone(id, id);
+
+            if (value == null)
+            {
+                e = new InvalidTimeZoneException(SR.Format(SR.InvalidTimeZone_InvalidFileData, id, AndroidTzDataInstance.GetTimeZoneDirectory() + TimeZoneFileName));
+                return TimeZoneInfoResult.TimeZoneNotFoundException;
+            }
+
+            e = null;
+            return TimeZoneInfoResult.Success;
+        }
+
+        private static string[] GetTimeZoneIds()
+        {
+            return AndroidTzDataInstance.GetTimeZoneIds();
+        }
+
+        /*
+        * Android v4.3 Timezone support infrastructure.
+        *
+        * This is a C# port of libcore.util.ZoneInfoDB:
+        *
+        *    https://android.googlesource.com/platform/libcore/+/master/luni/src/main/java/libcore/util/ZoneInfoDB.java
+        *
+        * This is needed in order to read Android v4.3 tzdata files.
+        *
+        * Android 10+ moved the up-to-date tzdata location to a module updatable via the Google Play Store and the
+        * database location changed (https://source.android.com/devices/architecture/modular-system/runtime#time-zone-data-interactions)
+        * The older locations still exist (at least the `/system/usr/share/zoneinfo` one) but they won't be updated.
+        */
+        private sealed class AndroidTzData
+        {
+            [StructLayout(LayoutKind.Sequential, Pack=1)]
```

Comment:
We no longer need `StructLayout`.

We also don't use the `signature` field anymore. This can now just be a regular struct with 3 int fields.

Reply:
Thanks, will reduce StructLayout to 2 int fields (removing finalOffset), and then looking to remove the struct all together. Will add comments about the assumptions of what the data header and data entries formats are so others can  understand the parsing.

Reply:
> Will add comments about the assumptions of what the data header and data entries formats are so others can understand the parsing.

+1. That sounds great. That way future readers can understand why certain bytes are being skipped.